app/models/manageiq/providers/microsoft/infra_manager.rb

```diff
@@ -24,8 +24,7 @@ def self.raw_connect(connect_params)
     require 'winrm'
 
     connect_params[:operation_timeout] = 1800
-    winrm = WinRM::Connection.new(connect_params)
-    winrm
+    WinRM::Connection.new(connect_params)
   end
```

Comment:
In particular I'm thinking if the purpose of getting a native handle is to get a shell, then that can be encapsulated into the `:service` parameter of the connect method.

So, roughly, a caller would do

```ruby
provider.with_provider_connection(:service => :powershell) do |shell|
  shell.run(...)
end
```

And the with_provider_connection or perhaps connect itself method would check for the service and auto-close the shell at the end of the block.